### ISSUE TITLE
FEATUREs: fifo support and colour support

### ISSUE THREAD

**Issue by BeyondMagic** [LoupLobet/sxd#1]
Hello, I don't know what's the current status of the project, but I've been using for some time and I'd love to ask for two features:

**The first**: the ability to change the text based on the input received on runtime.

For example, let's say that a script is reading from a fifo.

```bash
#!/usr/bin/env sh

read_fifo () {

  while read -r line; do

    echo "$line"

  done < /tmp/sxd_fifo

}

read_fifo
```

If we look at the output after running, `echo "read fifo" > /tmp/sxd_fifo`, we'll receive "read fifo".

Now, it would be awesome that sxd reading from this script could update the input as well.

---

**Secondly**: support for different colours based on the text received.

For example, lemonbar applies a different colours when we have `%{F#FFFFFF}White Text!%{F#000000}Black Text` while runtime, it would be nice to have a similar feature here as well.

---

I think those two features would be a neat to start working on independent widgets, simple, but powerful ways to show some type of information.

Of course, if it is that hard to add those supports, I'd start working on them on my own.

**Comment by LoupLobet** [comment]
Yes I had already asked myself the question of a fifo system to update the content of the window. It would be very great to add it to sxd (same for colour support). I would be happy to discuss about these additions in more detail with you. 

I'm currently very busy for the coming week. But I can already add you as a collaborator of the project (and we could start it within a week, if you're ok with that).

**Comment by BeyondMagic** [comment]
Oh, all right then. I am not really the best C programmer out there, like I feel I've not done enough with it, but I will definitely help with what I can when we start working on this.

Thank you for responding so fast and sorry for responding so late.

Question: next week like 7 days??

**Comment by LoupLobet** [comment]
Until Sunday 22 January
